### DISCUSSION TITLE
When the child component calls the parent component's state, the state of isIntersecting is reset

### DISCUSSION THREAD

**Discussion by condorheroblog** [thebuilder/react-intersection-observer#703] [General]
This link can be used to reproduce the scene: https://codesandbox.io/p/sandbox/isintersecting-is-reset-2xgfm3?file=%2Fsrc%2Findex.tsx%3A31%2C18

I updated the state of the parent component through the handleSetNum function in the child component.

<img width="544" alt="image" src="https://github.com/user-attachments/assets/0e00befb-32e8-4af9-b44f-2dc04efe92e0">

Print the value of `entry?.isIntersecting` when scrolling, and its value first changes to `true` and then to `false`. I don't know why.

![2024-09-11 17 32 03](https://github.com/user-attachments/assets/13f7edc6-61fb-42e6-933a-acfddc179ab4)

But if I don't use handleSetNum function everything is normal - one `true`.

![2024-09-11 17 33 01](https://github.com/user-attachments/assets/11136cb3-3579-4761-b9df-f820106386b2)

**Comment by thebuilder** [comment]
You have multiple observers that update the same context state in a `useEffect`.  You can only see two of the three, so you will have one that is `false`.

**Comment by condorheroblog** [comment]
I'm so sorry🤦‍♂️.

I forgot that there are three child components on my page.

Thank you.
